src/consumer.rs

```diff
@@ -0,0 +1,306 @@
+use crate::SharedFluvio;
+use crate::{optional_property, must_property};
+
+use log::debug;
+use flv_future_aio::task::spawn;
+use fluvio::PartitionConsumer;
+use fluvio::{Offset, FluvioError};
+use fluvio::dataplane::fetch::FetchablePartitionResponse;
+use fluvio::dataplane::record::RecordSet;
+
+use node_bindgen::derive::node_bindgen;
+use node_bindgen::core::NjError;
+use node_bindgen::core::JSValue;
+use node_bindgen::core::val::JsEnv;
+use node_bindgen::core::TryIntoJs;
+use node_bindgen::sys::napi_value;
+use node_bindgen::core::JSClass;
+use node_bindgen::core::val::JsObject;
+use node_bindgen::core::buffer::ArrayBuffer;
+
+pub struct PartitionConsumerWrapper {
+    client: SharedFluvio,
+    topic: String,
+    partition: i32,
+}
+
+impl PartitionConsumerWrapper {
+    pub fn new(client: SharedFluvio, topic: String, partition: i32) -> Self {
+        Self {
+            client,
+            topic,
+            partition,
+        }
+    }
+}
+
+impl TryIntoJs for PartitionConsumerWrapper {
+    fn try_to_js(self, js_env: &JsEnv) -> Result<napi_value, NjError> {
+        debug!("converting FluvioWrapper to js");
+        let new_instance = PartitionConsumerJS::new_instance(js_env, vec![])?;
+        debug!("instance created");
+        PartitionConsumerJS::unwrap_mut(js_env, new_instance)?.set_client(self.client);
+        PartitionConsumerJS::unwrap_mut(js_env, new_instance)?.set_topic(self.topic);
+        PartitionConsumerJS::unwrap_mut(js_env, new_instance)?.set_partition(self.partition);
+        Ok(new_instance)
+    }
+}
+
+pub struct PartitionConsumerJS {
+    inner: Option<SharedFluvio>,
+    topic: Option<String>,
+    partition: Option<i32>,
+}
+
+#[node_bindgen]
+impl PartitionConsumerJS {
+    #[node_bindgen(constructor)]
+    pub fn new() -> Self {
+        Self {
+            inner: None,
+            topic: None,
+            partition: None,
+        }
+    }
+
+    pub fn set_client(&mut self, client: SharedFluvio) {
+        self.inner.replace(client);
+    }
+
+    pub fn set_topic(&mut self, topic: String) {
+        self.topic.replace(topic);
+    }
+
+    pub fn set_partition(&mut self, partition: i32) {
+        self.partition.replace(partition);
+    }
+
+    #[node_bindgen]
+    async fn fetch(
+        &self,
+        offset: OffsetWrapper,
+    ) -> Result<FetchablePartitionResponseWrapper, FluvioError> {
+        let topic = self.topic.clone().unwrap_or_default();
+        let partition = self.partition.unwrap_or_default();
```

Comment:
I don't like this usage of `unwrap_or_default`, I think it can lead to silent errors.

Reply:
commit 0cc483b uses const defaults;